### PR TITLE
promises with aws.putObject

### PR DESCRIPTION
**Pull request by tomdavidson**
Very minor change. Promises on putObject just as with existing promises on getObject. Promises have been supported on putObject [since March](https://blogs.aws.amazon.com/javascript/post/Tx3BZ2DC4XARUGG/Support-for-Promises-in-the-SDK).

### CODE REVIEW

File: aws-sdk/aws-sdk.d.ts [hunk 1]
```diff
@@ -382,7 +382,7 @@ declare module "aws-sdk" {
     endpoint: Endpoint;
 
     getObject(params: s3.GetObjectRequest, callback?: (err: Error, data: any) => void): any;
-    putObject(params: s3.PutObjectRequest, callback: (err: Error, data: any) => void): void;
+    putObject(params: s3.PutObjectRequest, callback?: (err: Error, data: any) => void): any;
```
Comment (tomdavidson) [inline aws-sdk/aws-sdk.d.ts]: I like overload, but was trying to be consistent with s3.GetObjectRequest. I suggest another PR that refactors out the use of any return type just as you have described.

**Comment by vvakame** [comment]
We can't get response from author. please ping me after 3 days. I'll merge this PR.

**Comment by midknight41** [comment]
Agree with @vvakame, the overload is a nicer approach. Looking at the announcement from AWS, this change affects pretty much every method in definition file! Ouch.

**Comment by tomdavidson** [comment]
I agree the entire definition file could be refactored, but this PR was to fix a very tiny oversight and not overall code improvement so the same patterns that already exist were followed.

Refracting to use overloading is not mutually exclusive of this PR and I do not think it is justification to reject.

@vvakame merge?

**Comment by midknight41** [comment]
I wasn't suggesting you refactor the whole thing! That would be cruel and unusual punishment.

The any is is an improvement over broken so happy to take it. Any PR that improves this monster is appreciated. 

@vvakame: merge away.

**Comment by vvakame** [comment]
> I wasn't suggesting you refactor the whole thing!

:+1:

**Comment by tomdavidson** [comment]
@vvakame @midknight41  I agree, lets refactor but lets do it in a different pull request - #12205.

**Comment by vvakame** [comment]
okay, I'll waiting for that.

**Comment by tomdavidson** [comment]
Im not going to refactor due to AWS' plans to providing official typings and I closed #12205.  Can we merge this PR so we (I) dont have to use a forked version.

**Comment by midknight41** [comment]
:+1:

**Comment by midknight41** [comment]
Make sense to me. @vvakame can you do the honours?

**Comment by vvakame** [comment]
:+1:
but, conflict occured. could you resolve it?

**Comment by tomdavidson** [comment]
Lets get this merged before there is another conflict. If not, then lets just close the PR, I have committed my fork to my repos so my pipelines don't have to deal with it. 
```js
    "aws-sdk-types": "./bundled/aws-sdk-types",
```

**Pull request merged**